src/ILLink.RoslynAnalyzer/OperationExtensions.cs

```diff
@@ -0,0 +1,251 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.FlowAnalysis;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace ILLink.RoslynAnalyzer
+{
+	// Copied from Roslyn
+	internal static partial class OperationExtensions
+	{
+		/// <summary>
+		/// Returns the <see cref="ValueUsageInfo"/> for the given operation.
+		/// This extension can be removed once https://github.com/dotnet/roslyn/issues/25057 is implemented.
+		/// </summary>
+		public static ValueUsageInfo GetValueUsageInfo (this IOperation operation, ISymbol containingSymbol)
+		{
+			/*
+            |    code                  | Read | Write | ReadableRef | WritableRef | NonReadWriteRef |
+            | x.Prop = 1               |      |  ✔️   |             |             |                 |
+            | x.Prop += 1              |  ✔️  |  ✔️   |             |             |                 |
+            | x.Prop++                 |  ✔️  |  ✔️   |             |             |                 |
+            | Foo(x.Prop)              |  ✔️  |       |             |             |                 |
+            | Foo(x.Prop),             |      |       |     ✔️      |             |                 |
+               where void Foo(in T v)
+            | Foo(out x.Prop)          |      |       |             |     ✔️      |                 |
+            | Foo(ref x.Prop)          |      |       |     ✔️      |     ✔️      |                 |
+            | nameof(x)                |      |       |             |             |       ✔️        | ️
+            | sizeof(x)                |      |       |             |             |       ✔️        | ️
+            | typeof(x)                |      |       |             |             |       ✔️        | ️
+            | out var x                |      |  ✔️   |             |             |                 | ️
+            | case X x:                |      |  ✔️   |             |             |                 | ️
+            | obj is X x               |      |  ✔️   |             |             |                 |
+            | ref var x =              |      |       |     ✔️      |     ✔️      |                 |
+            | ref readonly var x =     |      |       |     ✔️      |             |                 |
+ 
+            */
+			if (operation is ILocalReferenceOperation localReference &&
```

Comment:
Why not to use switch instead?

Reply:
Copy-paste from Roslyn, I didn't change the code. Hopefully this will become a public API at some point.

Reply:
Could you please add a comment where the file was copied from